build/nlu-regression.js

```diff
@@ -0,0 +1,158 @@
+const fs = require('fs')
+const axios = require('axios').default
+const _ = require('lodash')
+
+const repoRootDir = `${__dirname}/..`
+const nluTestingDir = `${repoRootDir}/modules/nlu-testing/`
+
+const host = '127.0.0.1'
+const port = '3000'
+const base = `http://${host}:${port}`
+
+const login = async () => {
+  try {
+    const { data: login } = await axios.post(`${base}/api/v1/auth/login/basic/default`, {
+      email: 'admin',
+      password: '123456'
+    })
+    return login.payload.token
+  } catch {
+    return
+  }
+}
+
+const signup = async () => {
+  try {
+    const { data: login } = await axios.post(`${base}/api/v1/auth/register/basic/default`, {
+      email: 'admin',
+      password: '123456'
+    })
+    return login.payload.token
+  } catch {
+    return
+  }
+}
+
+const createBot = async (botId, axiosConfig) => {
+  const newBot = {
+    id: botId,
+    name: 'testy',
+    template: {
+      id: 'bp-nlu-regression-testing',
+      moduleId: 'nlu-testing'
+    },
+    category: undefined
+  }
+
+  try {
+    await axios.post(`${base}/api/v1/admin/bots`, newBot, axiosConfig)
+  } catch (err) {
+    const { status } = err.response
+    if (status === 409) {
+      console.log('bot already exists')
+      return
+    }
+    throw err
+  }
+}
+
+const waitForTraining = async (botId, axiosConfig) => {
+  return new Promise(function(resolve) {
+    let i = 0
+    console.log(`training...`)
+    const intervalId = setInterval(async () => {
+      const { data: trainingStatus } = await axios.get(`${base}/api/v1/bots/${botId}/mod/nlu/train`, axiosConfig)
+
+      if (!trainingStatus.isTraining) {
+        clearInterval(intervalId)
+        resolve()
+      } else {
+        console.log(`training... ${2 * ++i}s`)
+      }
+    }, 2000)
+  })
+}
+
+const runAllTests = async (botId, axiosConfig) => {
+  const baseNluTesting = `${base}/api/v1/bots/${botId}/mod/nlu-testing`
+  const { data: allTests } = await axios.get(`${baseNluTesting}/tests`, axiosConfig)
+  const nTests = allTests.length
+  let nPassing = 0
+
+  let i = 0
+  for (const test of allTests) {
+    const retry = async () => {
+      const { data } = await axios.post(`${baseNluTesting}/tests/${test.id}/run`, '', axiosConfig)
+      return data
+    }
+
+    let testResult
+    try {
+      testResult = await retry()
+    } catch (err) {
+      console.error(err, 'retrying')
+      testResult = await retry()
+    }
+
+    nPassing += testResult.success ? 1 : 0
+    console.log(`(${i++} /${nTests}) #${test.id}`, 'success: ', testResult.success)
+  }
```

Comment:
I'm a bit too used to typescript, can you avoid prefixing your variables with their type? Didn't test it, but I would suggest something like this 

```JS
const { data: tests } = await axios.get(`${baseNluTesting}/tests`, axiosConfig)

  const retry = async (testId) => {
    const { data } = await axios.post(`${baseNluTesting}/tests/${testId}/run`, '', axiosConfig)
    return data
  }

  let passedTests = 0
  await Promise.mapSeries(tests, async (test, idx) => {
    let testResult
    try {
      testResult = await retry(test.id)
    } catch (err) {
      console.error(err, 'retrying')
      testResult = await retry(test.id)
    }

    if(testResult.success){
      passedTests++
    }
    console.log(`(${idx} /${tests.length}) #${test.id}`, 'success: ', testResult.success)
  })
```